packages/user-operation-controller/src/UserOperationController.test.ts

```diff
@@ -824,6 +824,42 @@ describe('UserOperationController', () => {
       expect(prepareMock).toHaveBeenCalledTimes(1);
     });
 
+    it('uses gas limits suggested by snap during #addPaymasterData', async () => {
```

Comment:
This controller doesn't communicate with the snaps but with the `SmartContractAccount` abstraction so we can reference the `smart contract account` here.